eth2/types/src/slot_epoch_macros.rs

```diff
@@ -192,17 +192,31 @@ macro_rules! impl_display {
 
 macro_rules! impl_ssz {
     ($type: ident) => {
-        impl Encodable for $type {
-            fn ssz_append(&self, s: &mut SszStream) {
-                s.append(&self.0);
+        impl Encode for $type {
+            fn is_ssz_fixed_len() -> bool {
+                <u64 as Encode>::is_ssz_fixed_len()
```

Comment:
These trait annotations are necessary because the methods of `Decode` and `Encode` have the same names? I'm OK with that tradeoff I guess

Reply:
That's correct. An alternative is to provide enc/dec via the same trait, I figured this is more modular. We could also give the traits different names but that seemed a bit verbose. Open to suggestions.

Reply:
I think this strikes the right balance, having the method names the same is quite nice, and the trait annotations aren't too onerous